### ISSUE TITLE
Hook to publish release

### ISSUE THREAD

**Issue by chrisharris77** [netzwerg/gradle-release-plugin#31]
Is there any way to build and publish release artifacts: 

- build snapshot
- update version
- commit version
- push version
- tag version
- **build release**
- **publish release**
- update version
- commit version
- push version

**Comment by netzwerg** [comment]
Calling the `release` task will not build a snapshot, but build a release, hence your **build release** scenario is covered.

Concerning **publish release**: This is out of this plugin's scope. The recommended approach depends on your definition of "publish":

* For Artifactory publishing, I recommend the [Gradle Artifactory Plugin](https://www.jfrog.com/confluence/display/RTF/Gradle+Artifactory+Plugin)
* For GitHub publishing, you might want to look into my `gradle-release-pub-plugin`. It is two years old and very abandoned (there was no real interest), but it might serve as an inspiration.
* ...

Hope this helps – Cheers, Rahel

**Comment by chrisharris77** [comment]
Thank you Rahel. I completely agree it's not a release plugin's job to perform a publish.

I am using maven-publish plugin. But the problem I have is that there is no way to invoke the publish plugin at the right time in the middle of the release i.e. when version.txt contains the release version number. I can only publish before or after the whole release task runs - which results in publishing a snapshot not a release. 

Regards,
Chris

**Comment by netzwerg** [comment]
Yep, your are right, there are currently no hooks to perform work in between the "tag version" and "update version" steps.

However, you could perform a `git checkout <tag>` right after the completion of `release`, and then call `build` and `publish` afterwards – could that work for you?

**Comment by chrisharris77** [comment]
I suppose that would be a workaround if I can get the tag.

It would be much nicer to not have to drop to executing git commands though. It seems like doing work when the version is set to the release version would be a very common scenario. What about making the release task be a wrapper for sub tasks? Then users could choose to the release task as is, or invoke subtasks and mix with other tasks e.g. publish?

**Comment by netzwerg** [comment]
The tag is exposed via `project.release.tagName`.

I completely agree that we should offer cleaner solution – I am very open to pull requests. The suggested approach is really only a workaround, just so you are not blocked until a better solution is available.

**Comment by chrisharris77** [comment]
OK I'll try that thanks.
